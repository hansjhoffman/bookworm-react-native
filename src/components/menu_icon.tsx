import React from "react";
import { Image, StyleSheet, TouchableOpacity } from "react-native";

import { Colors, Layout } from "../constants";
import { Icons } from "../../assets";

type Props = {
  onPress?: () => void;
};

export function MenuIcon({ onPress }: Props) {
  return (
    <TouchableOpacity onPress={onPress} testID="menu-icon">
      <Image style={styles.menuIcon} source={Icons.nav_menu} />
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  menuIcon: {
    width: 23,
    height: 23,
    marginLeft: Layout.margin_lg,
    resizeMode: "contain",
    tintColor: Colors.iconSelected,
  },
});
