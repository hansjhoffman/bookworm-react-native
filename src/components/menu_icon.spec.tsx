import React from "react";
import { fireEvent, render } from "@testing-library/react-native";

import { MenuIcon } from "./menu_icon";

describe("MenuIcon", function() {
  it("should handle onPress if given", function() {
    const onPress = jest.fn();

    const { getByTestId } = render(<MenuIcon onPress={onPress} />);

    fireEvent.press(getByTestId("menu-icon"));

    expect(onPress).toHaveBeenCalled();
  });

  it("should match snapshot", function() {
    const { baseElement } = render(<MenuIcon />);

    expect(baseElement).toMatchSnapshot();
  });
});
