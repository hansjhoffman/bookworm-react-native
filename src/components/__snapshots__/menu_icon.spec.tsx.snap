// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] MenuIcon should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      testID="menu-icon"
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/icons/nav_menu.png",
          }
        }
        style={
          Object {
            "height": 23,
            "marginLeft": 20,
            "resizeMode": "contain",
            "tintColor": "#386782",
            "width": 23,
          }
        }
      />
    </TouchableOpacity>
  </View>
</View>
`;
